tokio/src/process/unix/mod.rs

```diff
@@ -118,12 +121,24 @@ pub(crate) fn spawn_child(cmd: &mut std::process::Command) -> io::Result<Spawned
     let stdout = child.stdout.take().map(stdio).transpose()?;
     let stderr = child.stderr.take().map(stdio).transpose()?;
 
+    #[cfg(target_os = "linux")]
+    match pidfd_reaper::PidfdReaper::new(child, GlobalOrphanQueue) {
```

Comment:
Are there tests today that check this code path? Do we test spawning children both from the runtime and outside of the runtime?

Reply:
Test for `PidfdReaper` does not check for creation of it outside tokio runtime.

It uses `io::PollEvent`, so I assume it will panic if not in runtime.

Reply:
`PidfdReaper::new` calls `PollEvented::new_with_interest` , which calls `runtime::scheduler::Handle::current()` to get thje handle to current scheduler.

It panics if there's no scheduler, since `PollEvented` needs to hold a reference to scheduler, I'm not sure how to work around this cc

Reply:
`PidfdReaper` does have a unit test similar to `Reaper<_, _, _>` and there are integration testings in `tokio/tests/process*.rs`

Reply:
Looking at the path for the `Reaper`, I realized that this is not a problem since `Reaper::new` requires `signal::unix::signal()` to be called to obtain the signal handler, which also calls `runtime::scheulder::Handle::current()`